packages/api/cms-api/src/user-permissions/user-permissions.types.ts

```diff
@@ -0,0 +1,35 @@
+import { FindUsersArgs } from "./dto/paginated-user-list";
+import { User } from "./dto/user";
+import { UserPermission } from "./entities/user-permission.entity";
+import { ContentScope } from "./interfaces/content-scope.interface";
+import { Permission } from "./interfaces/user-permission.interface";
+
+export type PermissionConfiguration = Record<string, unknown>;
+
+export enum USERPERMISSIONS {
+    allContentScopes = "all-content-scopes",
+    allPermissions = "all-permissions",
+}
+
+type UserPermissions =
+    | Pick<UserPermission, "permission" | "configuration" | "validFrom" | "validTo" | "reason" | "requestedBy" | "approvedBy">[]
+    | USERPERMISSIONS.allPermissions;
+
+export type ContentScopeValues = {
+    [key in keyof ContentScope]: ContentScope[key][];
+};
+
+export type UserContentScopes = ContentScopeValues | USERPERMISSIONS.allContentScopes;
+
+export const USERPERMISSIONS_CONFIG = "userpermissions-config";
```

Comment:
There should be an underscore between USER and PERMISSIONS

Reply:
You could also move constants into a separate constants file

Reply:
https://github.com/vivid-planet/comet/pull/1223/commits/9272772317cfd1a27a08bcee65c3b59af5f4658c